scripts/get-version.ps1

```diff
@@ -4,17 +4,42 @@ $ErrorActionPreference="Stop"
 git update-index -q --refresh
 git diff-files --quiet | Out-Null
 
-$dirtyTag=""
-if ($LASTEXITCODE -ne 0) {
-    $dirtyTag = "-dirty"
-}
+$dirty=($LASTEXITCODE -ne 0)
 
 try { 
   git describe --tags --exact-match >$null 2>$null
   # If we get here the above did not throw, so we can use the exact tag
-  Write-Output "$(git describe --tags --exact-match)$dirtyTag"
+  if ($dirty) {
+      Write-Output "$(git describe --tags --exact-match)"
+  } else {
+      Write-Output "$(git describe --tags --exact-match)+dirty"
+  }
 } catch {
-  # Otherwise, append the timestamp of the commit and the hash
-  Write-Output "$(git describe --tags --abbrev=0)-$(git show -s --format='%ct-g%h')$dirtyTag"
+    # Otherwise, take the existing tag, increment the patch version and append the timestamp of the commit and hash
+    $tag=""
+
+    try {
+        git describe --tags --abbrev=0 >$null 2>$null
+        $tag="$(git describe --tags --abbrev=0)"
+    } catch {
+        $tag="v0.0.0"
```

Comment:
Probably a stupid question, but is `0.0.0` valid semver?

Reply:
It is (at least the `semver` node package doesn't complain about it), but also note that we'll increment the patch number a few lines later, so the version we'll generate will be `0.0.1-dev.XXXXXX`